instrumentation/http-url-connection/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/httpurlconnection/SunHttpUrlHttpAttributesGetter.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.httpurlconnection;
+
+import io.opentelemetry.instrumentation.api.instrumenter.http.HttpClientAttributesGetter;
+import java.net.HttpURLConnection;
+import java.util.List;
+import javax.annotation.Nullable;
+
+class SunHttpUrlHttpAttributesGetter
+    implements HttpClientAttributesGetter<HttpURLConnection, Integer> {
+
+  private final HttpUrlHttpAttributesGetter delegate = new HttpUrlHttpAttributesGetter();
+
+  @Override
+  public String method(HttpURLConnection connection) {
+    String requestMethod = connection.getRequestMethod();
+
+    if ("GET".equals(requestMethod)) {
```

Comment:
Did you consider doing this inside `HttpUrlHttpAttributesGetter`?
If you did it this way because there isn't an obvious way how to tell that `getOutputStream` was called in `HttpUrlHttpAttributesGetter` then heres a couple of ideas for you. You could try using `HttpUrlState` add a boolean field to it that would signify that it really is a `POST`. Another way would be to use `addContextCustomizer` method from `InstrumenterBuilder`. The idea is to use `addContextCustomizer` to add a holder object into context where you can store info like is this really a `POST`.  For example see https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/grizzly-2.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/grizzly/GrizzlyErrorHolder.java. As you don't have access to context in `HttpUrlHttpAttributesGetter.method` you'll need to add a  custom `AttributesExtractor` to your `Instrumenter`. Just leave `HttpUrlHttpAttributesGetter` as is and overwrite the request method attribute in your `AttributesExtractor.onEnd`. If you set the `SemanticAttributes.HTTP_METHOD` attribute in `onEnd` then it will work even when the span wasn't started from `getOutputStream`.

Reply:
Thank you for your suggestions  @laurit and @mateuszrzeszutek.  In my last commit, I created a new implementation to not depend on the execution order of the connection's methods.